### PR TITLE
Add more robust aspect ratio handling

### PR DESCRIPTION
**Pull request by Hoikas**
Ultrawide displays are generally advertised as having a 21:9 aspect ratio, even when they're a bit off from that mathematically speaking. So, we now allow the text for the aspect ratio to be set separate from the numerical values. Also added some additional fudging to handle all the oddball resolutions that DirectX reports.

### CODE REVIEW

File: Python/xOptionsMenu.py [hunk 1]
```diff
@@ -1540,13 +1554,12 @@ def InitVideoControlsGUI(self):
...
     def _AspectRatio(self, w, h):
         """Returns the appropriate aspect ratio for the given resolution"""
-        ratios = ((5, 4), (4, 3), (3, 2), (16, 10), (5, 3), (16, 9), (16, 9.375),)
-        w = float(w) # comes in as string, want float (not int) for division
-        h = float(h)
-        for r in ratios:
-            # resolution is within 1 pixel wiggle room in any direction from the exact aspect ratio (needed to recognize 1280x854 as 3:2)
-            if (w+1)/(h-1) >= float(r[0])/float(r[1]) >= (w-1)/(h+1):
-                return " [%i:%i]" % (r[0], r[1])
+        # allow some wiggle room for float comparisons (CWalter particularly points out 1280x854)
```
Comment (cwalther) [inline Python/xOptionsMenu.py]: Nitpicks:
• I prefer to have my name spelled with an h.
• The wiggle room wasn’t primarily for float comparison (although that was a welcome side effect), but exactly for matching aspect ratios that aren’t mathematically precise (854*3/2 = 1281, not 1280). Obviously you have now found ones that are even further off than what I tested at the time. (So maybe there’s no need for dealing with pixels anymore and one could just accept a certain range of ratios.)

Comment (Hoikas) [reply Python/xOptionsMenu.py]: Whoops! Sorry about the typo in your name--I will fix that.

I do realize why you had the wiggle room -- to me, float comparison and mathematically imprecise resolutions are symptoms of the same problem. It seemed to me that aspect ratios at most were expressed to the hundredths place in various online fora, so I went with that much precision.

I did run some calculations and saw some edge cases that the wiggle room would catch that my stupid rounding would not, however. When they both act together, all the resolutions offered on my ultrawide display have an aspect ratio listed. I suppose I could delete the wiggle room and see if that causes any issues.

**Comment by Hoikas** [comment]
Updated to address concerned by @CWalther. I arrived at my aspect ratio ranges by some liberal futzing with the values found on https://en.wikipedia.org/wiki/List_of_common_resolutions

**Comment by cwalther** [comment]
Thanks for the name correction, and yay for shorter code!
While we’re bikeshedding:
• Changing to `"4:3": (1.28, 1.35)` and `"16:9": (1.70, 1.78)` would make it additionally recognize 1024x600, 1024x800, 1120x832 from that list. (Anyone playing on a NeXT?)
• The round() is not really needed, you could absorb its effect into the tabulated bounds by adjusting them by 0.005, but that’s not even necessary, they work fine as they are.
• Any reason _ratios is a dictionary when you never do a lookup in it?
• “beging marketed”?

**Comment by Hoikas** [comment]
Good points. I've pushed changes addressing all of the above. I noticed when removing the `round()` that, on my machine, this affected the oddball 1768x992 resolution. I fixed this by adding another hundredth of fudge to the 16:9 aspect ratio range maximum.

**Pull request merged**